processor/src/main/java/org/mapstruct/ap/spi/DefaultAccessorNamingStrategy.java

```diff
@@ -68,36 +79,103 @@ private boolean isGetterMethod(ExecutableElement method) {
     }
 
+
+    /**
+     * Returns true when the {@link ExecutableElement} is a setter method. A setter starts with 'set'. The remainder of
+     * the name is supposed to reflect the property name.
+     *
+     * MapStruct will ensure that there's only one argument.
+     *
+     * @param method to be analyzed
+     * @return true when the method is a getter.
+     */
     public boolean isSetterMethod(ExecutableElement method) {
         String methodName = method.getSimpleName().toString();
 
         return methodName.startsWith( "set" ) && methodName.length() > 3;
     }
 
+    /**
+     * Returns true when the {@link ExecutableElement} is a setter method. An adder method starts with 'add'. The
+     * remainder of the name is supposed to reflect the property name.
+     *
+     * MapStruct will ensure that there's only one argument.
+     *
+     * @param method to be analyzed
+     * @return true when the method is an adder method.
+     */
     public boolean isAdderMethod(ExecutableElement method) {
         String methodName = method.getSimpleName().toString();
 
         return methodName.startsWith( "add" ) && methodName.length() > 3;
     }
 
+    /**
+     * Returns true when the {@link ExecutableElement} is a setter method. An presence check method starts with
+     * 'has'. The remainder of the name is supposed to reflect the property name.
+     *
+     * MapStruct will ensure that there's only one argument and that the return type is boolean or a {@link Boolean}
+     *
+     * @param method to be analyzed
+     * @return true when the method is a presence check method.
+     */
+    public boolean isPresenceCheckMethod(ExecutableElement method) {
+        String methodName = method.getSimpleName().toString();
+        return methodName.startsWith( "has" ) && methodName.length() > 3;
+    }
 
+    /**
+     * Analyzes the method (getter or setter) and derives the property name.
+     * See {@link #isGetterMethod(ExecutableElement)} {@link #isSetterMethod(ExecutableElement)}. The first three
+     * ('get' / 'set' scenario) characters are removed from the simple name, or the first 2 characters ('is' scenario).
+     * From the remainder the first character is made into small case (to counter camel casing) and the result forms
+     * the property name.
+     *
+     * @param getterOrSetterMethod getter or setter method.
+     *
+     * @return the property name.
+     */
     @Override
     public String getPropertyName(ExecutableElement getterOrSetterMethod) {
         String methodName = getterOrSetterMethod.getSimpleName().toString();
         return Introspector.decapitalize( methodName.substring( methodName.startsWith( "is" ) ? 2 : 3 ) );
     }
 
+    /**
+     * Adder methods are used to add elements to collections on a target bean. A typical use case is JPA. The
+     * convention is that the element name will be equal to the remainder of the add method. Example: 'addElement'
+     * element name will be 'element'.
```

Comment:
Ah, okay... perhaps add a `{@link ..} to this method above in the isAdderMethod doc.